ashpd-demo/src/portals/desktop/usb.rs

```diff
@@ -0,0 +1,362 @@
+                                    }
+                                }
+                            }
+                        ));
+                    }
+                ));
+                let device_id = device.0.clone();
+                row.connect_unshare_clicked(clone!(
+                    #[strong]
+                    page,
+                    move |_row| {
+                        glib::spawn_future_local(clone!(
+                            #[strong]
+                            device_id,
+                            #[strong]
+                            page,
+                            async move {
+                                let usb = UsbProxy::new().await.unwrap();
+                                let result = usb.release_devices(&[&device_id]).await;
+                                println!("{result:?}");
+                                match result {
+                                    Ok(_) => {}
+                                    Err(err) => {
+                                        tracing::error!("Acquire device error: {err}");
+                                        page.error(&format!("Acquire device error: {err}"));
+                                    }
+                                }
+                                page.imp().released_device(&device_id);
+                            }
+                        ));
+                    }
+                ));
+                page.imp().add(device.0.clone(), row);
+            }
+            Ok(())
+        }
+
+        pub(super) async fn start_session(&self) -> ashpd::Result<()> {
+            let usb = UsbProxy::new().await?;
+            let session = usb.create_session().await?;
+            self.session.lock().await.replace(session);
+
+            let session = self.session.clone();
+            glib::spawn_future(async move {
+                let usb = UsbProxy::new().await?;
+                loop {
+                    if session.lock().await.is_none() {
+                        tracing::debug!("session is gone");
+                        break;
+                    }
+                    if let Some(response) = usb.receive_device_events().await?.next().await {
+                        let events = response.events();
+                        for ev in events {
+                            println!(
+                                "Received event: {} for device {}",
+                                ev.event_action(),
+                                ev.event_device_id()
+                            );
+                        }
+                    }
+                }
+                tracing::debug!("Loop is gone");
+                Ok::<(), ashpd::Error>(())
+            });
+
+            Ok(())
+        }
+
+        pub(super) async fn stop_session(&self) -> anyhow::Result<()> {
+            if let Some(session) = self.session.lock().await.take() {
+                session.close().await?;
```

Comment:
should switch the buttons state

Reply:
this is done by the non-imp version which is the one called from the UI.